### ISSUE TITLE
not working for me on phpbb 3.3.3

### ISSUE THREAD

**Issue by DjDecibel** [Digitalroot/MediaWiki_PHPBB_Auth#43]
I just installed mediawiki 1.35.1, installed the PluggableAuth 1.35 and your 4.0.0 alpha 2 version of phpBB_Auth

my phpBB version is 3.3.3

when i try to access with the correct php accounts (tested them and working on phpBB) i always get the error:

Please register on the forums to login.
<a href="https://www.****.com/Forums/ucp.php?mode=register">Click here to create an account.</a>

i also don't know why it also show the html tag instead of the link..

btw it doesn't work for me..

I did some testing trying to login with wrong username and the same error appears!

**Comment by codymatthew** [comment]
Have you gotten this to work at all?

**Comment by DjDecibel** [comment]
no, still not working for me.. :\

**Comment by rogalian** [comment]
Also not working on 3.3.4 (updated today, inc Pluggable Auth)
Internal error
[YKWPq8H0yLjeNsBe4YaQHAAAAAk] 2021-05-19 22:22:35: Fatal exception of type "Error"

Previous combination of Auth_remoteuser and phpbbauth continues to work.

**Comment by cpeel** [comment]
I have this working with:
* MediaWiki_phpBB v4.0.0-alpha.2
* MediaWiki 1.35
* PluggableAuth efff551 released on 2021-12-13T07:26:17
* phpBB 3.3.3
* PHP 7.4.3

I'm interested in helping folks who are having problems. Please start by providing the versions of the above software you are using and I'll see if I can repro it.

**Comment by JWPlatt** [comment]
@cpeel Does your testing include confirming that phpBB-to-MediaWiki username translation works?

**Comment by cpeel** [comment]
I'm not sure what you're asking. phpBB users without a MediaWiki account are able to log into MediaWiki using their phpBB creds and an account is created for them within MediaWiki that matches their phpBB username (with an uppercase first letter, per MediaWiki guidelines). My site uses `$wgAuth_Config['UseCanonicalCase'] = true;` and I have confirmed the case is retained.

**Comment by Digitalroot** [comment]
@cpeel  He might be asking about non-latin languages. I recall the plug-in being used by a lot of non-English speakers. The plug-in uses something called case folding. 

If you have any PRs I will gladly accept them. Unfortunately my development path has had me doing C# and not PHP. I do not even use MediaWiki or phpBB anymore.

**Comment by JWPlatt** [comment]
@cpeel @Digitalroot I will then assume the entire package has not been tested.  Please see README.md on the code page under "Optional Features."  I was its author.  It was a significant addition and any unawareness surprises me.  The problem was that there was no automated way to reliably translate phpBB usernames in a satisfying way when they conflict with what MediaWiki allows.

UseCanonicalCase is not enough.  There are characters phpBB allows that MediaWiki does not.  My "phpBB-to-MediaWiki username translation" addition to the package provides a custom phpBB profile field, restricted to admins, to translate a phpBB username to a more restrictive MediaWiki username.  For example, phpBB allows "My_User_Name" whereas MediaWIki does not allow the underscores.  So the user might call themself "MyUserName" on the wiki and ask the admin to enter that into phpBB under his phpBB account profile.

**Comment by Digitalroot** [comment]
Thanks for the reminder, I am recalling that now.

**Comment by cpeel** [comment]
We are not using the optional mapping feature. The way we use phpBB we restrict what characters are used in usernames for other reasons and thus `UseCanonicalCase` is sufficient for us.

That gives me the information I need for further testing and I will do so.

**Comment by cpeel** [comment]
@JWPlatt - I've figured out how I broke `UseWikiProfile` in the conversion to PluggableAuth and I know largely how to fix it, but to make sure I get it working as intended I have some clarifying questions on how this is suppose to work:
1. Are users expected to log into the wiki with their phpBB username or the wikiusername set for them in the extra field?
2. Is the intent that if a user has a wikiusername value in the database then it will be used as their MediaWiki username regardless if their phpBB username is a valid MediaWiki username?

**Comment by JWPlatt** [comment]
@cpeel Thanks for your work.
1. Users expect to log into the wiki with their wikiusername and into phpBB with their phpBB username, using their phpBB password for both.  Otherwise, if they had to use their phpBB account name on the wiki, they'd lose their original wiki account and edit history when integrated midstream on an operating wiki.  I wrote the mod in 2017, so I hope I'm remembering right.
2. The intent is to translate between the phpBB username and a valid wiki username and authenticate against phpBB to use the better security controls in phpBB, to stop wiki spam, with wiki edit permission controlled by membership in a phpBB open usergroup.  Usernames that are the same and valid on both phpBB and the wiki, and owned by the same person, do not need a translating entry.  It could be used, for example, to translate usernames that are quite different, or are coincidentally the same on phpBB and the wiki before integration but owned by different people.  It is for this reason that only an admin can enter a translation from a phpBB account to a wiki account.

I hope that helps.

**Comment by cpeel** [comment]
Thanks for that clarification and I should have a PR for a fix out here shortly.

I'm glad I asked because I would have expected users to log in their phpBB username and the system use the wikiusername. The reason is that for us at Distributed Proofreaders, wiki accounts are only created by authenticating against phpBB so there is no other way to create accounts. Honestly, it should be easy enough to make this work both ways defaulting to the original design that uses the wikiusername first.

**Comment by cpeel** [comment]
I've opened https://github.com/Digitalroot-Technologies/MediaWiki_PHPBB_Auth/pull/46 to correctly handle `UseWikiProfile`.

I intend a separate PR to add some `wfDebugLog()` statements that we can enable to help diagnose problems. I also will open a separate PR with some `README.md` updates.

**Comment by cpeel** [comment]
For users who are experiencing problems: I recommend waiting until https://github.com/Digitalroot-Technologies/MediaWiki_PHPBB_Auth/pull/50 is merged in and then trying again and providing feedback. `UseWikiProfile` is now fixed and the mentioned PR implements more robust error reporting to help with configuration and authentication issues.

**Comment by cpeel** [comment]
PR 50 has merged. If you are experiencing problems with the latest 4.x code, enable the debug log as indicated in the [troubleshooting section](https://github.com/Digitalroot-Technologies/MediaWiki_PHPBB_Auth#troubleshooting) of the `README.md`.

Due to how the 4.x plugin integrates with the MediaWiki authentication system, login messages must be formatted using wiki markup. In particular links will not render correctly unless wiki link formats are used:
```php
$wgAuth_Config['LoginMessage']   = "'''Please register on the forums to login.''' [https://some.domain.com/phpBB/ucp.php?mode=register Click here to create an account.]";
```

**Comment by JWPlatt** [comment]
@cpeel  Thank you - looks good.  The original development of the phpBB wiki group/profile username translation feature took a couple of weeks at least and I just did not have the desire to do that research and development again.  It's rewarding that you not only brought it back into compliance, but also improved ease of use and expanded upon it.  Thanks again.

**Comment by Digitalroot** [comment]
I think this is resolved w/ #51
